fix: remove leftover elements after AI content generation

syuzanvs opened the pull request:
- Improve selection handling in aiagentservice.ts by checking if selection spans entire blocks
- Remove insertEmptySpace function calls that were adding unwanted non-breaking spaces
- Fix content processing to properly handle block-level selections without extra whitespace
- Replace selectAll + insertContent with direct model manipulation for cleaner content replacement

This resolves issue where AI agent would insert unwanted spaces and newlines when editing selected text, particularly when the selection spans entire blocks.

Closes #154

jjroelofs reviewed (changes requested):
On this branch I am still able to reproduce the problem intermittently:

https://app.screencastify.com/watch/CAMQEalwq2b6c9UuoddN?checkOrg=e05d881a-3bd2-4c6f-b728-427fbaef3812

Can we prevent these spaces being added? A single space is added every time we include the generation animation icon, can't we do that without showing a visible spacer?

In the cases where I'm reproducing this, I wonder if we need to trim the AI output at some point.

jjroelofs reviewed (changes requested):
new regression bug: If I edit a heading it is not a heading anymore. This was not reproducible on the main branch.

https://app.screencastify.com/watch/GCRlWqiXHmAZckwGZtHb?checkOrg=e05d881a-3bd2-4c6f-b728-427fbaef3812

syuzanvs committed:
fix: remove leftover elements after AI content generation

- Improve selection handling in aiagentservice.ts by checking if selection spans entire blocks
- Remove insertEmptySpace function calls that were adding unwanted non-breaking spaces
- Fix content processing to properly handle block-level selections without extra whitespace
- Replace selectAll + insertContent with direct model manipulation for cleaner content replacement

This resolves issue where AI agent would insert unwanted spaces and newlines when editing selected text, particularly when the selection spans entire blocks.

Closes #154

jjroelofs reviewed (approved):
Perfect! Im glad we got rid of the non-breaking-space that we inserted for the loader graphic that was not necessary. This makes editing with AI feel tighter and more stable.
